go_modules/lib/dependabot/go_modules/update_checker.rb

```diff
@@ -16,7 +16,9 @@ def latest_resolvable_version
         # To update indirect dependencies we'll need to promote the indirect
         # dependency to the go.mod file forcing the resolver to pick this
         # version (possibly as # indirect)
-        return dependency.version unless dependency.top_level?
+        unless dependency.top_level?
+          return version_class.new(dependency.version)
```

Comment:
Think the version can be nil, do we need to handle this when initializing the version class?

Reply:
The version class doesn't support it, when can this be nil exactly? I can add test coverage for it and handle that case 👍

Reply:
I've added a naive `return unless dependency.version`, since I'm not sure how this happens exactly, but figured better safe than sorry